tests/constraints/tests.py

```diff
@@ -891,8 +891,17 @@ def test_model_validation_with_condition(self):
             name=obj1.name, color="blue"
         ).validate_constraints()
         msg = "Constraint “name_without_color_uniq” is violated."
-        with self.assertRaisesMessage(ValidationError, msg):
+        with self.assertRaisesMessage(ValidationError, msg) as cm:
             UniqueConstraintConditionProduct(name=obj2.name).validate_constraints()
+            self.assertEqual(cm.exception.error_dict["__all__"][0].code, "other_code")
+
+        msg = "Custom message"
+        with self.assertRaisesMessage(ValidationError, msg) as cm:
+            UniqueConstraintProduct(
+                color=self.p1.color,
+                age=self.p1.age,
+            ).validate_constraints()
+        self.assertEqual(cm.exception.error_dict["__all__"][0].code, "custom_code")
```

Comment:
e.g. the `color_age_uniq` constraint definition could be inlined in the test and have it's `validate` method directly called.